server/src/domain/system-config/system-config.service.ts

```diff
@@ -53,6 +73,48 @@ export class SystemConfigService {
     this.core.addValidator(validator);
   }
 
+  async handleImmichLatestVersionAvailable() {
+    if (this.disableCheckLatestVersion) {
+      return true;
+    }
+    const proxy =
+      this.proxyHost && this.proxyPort && this.proxyProtocol
+        ? {
+            protocol: this.proxyProtocol,
+            port: this.proxyPort,
+            host: this.proxyHost,
+          }
+        : false;
+
+    try {
+      const { data } = await axios.get<GithubRelease>(
+        'https://api.github.com/repos/immich-app/immich/releases/latest',
+        {
+          headers: {
+            Accept: 'application/vnd.github.v3+json',
+          },
+          proxy,
```

Comment:
Why is it necessary to expose `proxy` settings here? I think there might be out of the box environment variables to configure this automatically `http_proxy` and `https_proxy`.

Reply:
What's the best between configuring the proxy in the Immich system settings and with an environment variable ?

Reply:
I'd say the UI whenever possible. If we're going to add a proxy thing, can we make it a single string/url?